### ISSUE TITLE
time series issue with ifort built MF6

### ISSUE THREAD

**Issue by ougx** [MODFLOW-USGS/modflow6#1439]
**Describe the bug**
The ifort built MF6 does not correctly assign values in the time series data file. In the attached example, the SFR inflows for time 8 and 10 were incorrectly assigned when running a ifort built MF6. However, gfortran built MF6 does not have this problem.
[step1.zip](https://github.com/MODFLOW-USGS/modflow6/files/13325868/step1.zip)

**Comment by wpbonelli** [comment]
thanks for reporting @ougx just to confirm, you are seeing this with the latest release, develop branch, or both?

**Comment by ougx** [comment]
Both

**Comment by wpbonelli** [comment]
I also see 7 and 9 at t=8 and 10 respectively on mac 13.6.2 ifort 2021.10, building from develop

**Comment by ougx** [comment]
It turned out to be the mixed new line characters used in the flow file. Strange that ifort could not handle this.

**Comment by wpbonelli** [comment]
We should probably document this as a limitation of the Intel-built distributions (I do not see it mentioned in mf6io) and test whether ifx has the same issue

**Comment by ougx** [comment]
FYI, the file can actually be read correctly by ifort using free format `*` below. 

```fortran
program readfile
double precision :: time, flow
integer          :: i, ifile


open(newunit=ifile, file='ts_timestep.sfr_flow')
do i=1, 7
  read(ifile, *)
end do
do i=0,14
  read(ifile, *) time, flow
  print*, time, flow
end do
end
```

**Comment by wpbonelli** [comment]
As @emorway-usgs suggests this could stay open until documented or fixed

**Comment by langevin-usgs** [comment]
We have a special line reader in mf6 that allows us to read lines with unlimited length.  Apparently, it needs some additional work to handle mixed line returns.  Very strange that the behavior is compiler dependent.

**Comment by vivekbedekar** [comment]
I have added code to trap this error in the get_line subroutine. Will run autotest and then check in the code.

**Comment by langevin-usgs** [comment]
Fixed with #1866
